contracts/cw1-subkeys/src/contract.rs

```diff
@@ -505,714 +651,1060 @@ mod tests {
         )
     }
 
-    #[test]
-    fn query_allowance_works() {
-        let mut deps = mock_dependencies(&[]);
-
-        let owner = "admin0001";
-        let admins = vec![owner, "admin0002"];
-
-        let spender1 = "spender0001";
-        let spender2 = "spender0002";
-        let spender3 = "spender0003";
-        let initial_spenders = vec![spender1, spender2];
-
-        // Same allowances for all spenders, for simplicity
-        let denom1 = "token1";
-        let amount1 = 1111;
-
-        let allow1 = coin(amount1, denom1);
-        let initial_allowances = vec![allow1.clone()];
-
-        let expires_never = Expiration::Never {};
-        let initial_expirations = vec![expires_never, expires_never];
-
-        let info = mock_info(owner, &[]);
-        setup_test_case(
-            deps.as_mut(),
-            &info,
-            &admins,
-            &initial_spenders,
-            &initial_allowances,
-            &initial_expirations,
-        );
+    mod allowance {
+        use super::*;
+
+        #[test]
+        fn query() {
+            let Suite { deps, .. } = SuiteConfig::new()
```

Comment:
I'm loving the API design here.
Super concise and flexible. The builder-pattern SuiteConfig is working out very well. Much better than a number of setup functions with different args... so much more power here.